File: Leetcode/myPow.swift
func myPowI(x: Double, _ n: Int) -> Double {
    
    if n == 0 {
        return 1.0
    } else if x == 0.0 {
        return 0.0
    } else if x == 1.0 {
        return 1.0
    } else if n == 1 {
        return x
    } else {
        if n < 0 {
            return 1.0 / myPowI(x, -n)
        }
        var half: Double = myPowI(x, n / 2)
        if n % 2 == 0 {
            return half * half
        } else {
            return half * half * x
        }
    }

}

func myPowII(x: Double, _ n: Int) -> Double {
    if n == 0 {
        return 1.0
    } else if x == 0.0 {
        return 0.0
    } else if x == 1.0 {
        return 1.0
    } else if n == 1 {
        return x
    } else {
        var isNeg: Bool = n < 0
        var m: Int = n
        if n < 0 {
            m = -m
        }
        var p: Double = x
        var result: Double = 1.0
        while m > 0 {
            if (m & 1) != 0 {
                result *= p
            }
            p *= p
            m = m >> 1
        }
        return isNeg ? 1.0 / result : result
        
    }
}

File: Leetcode/myPow.java
public double myPow(double x, int n) {
        if (n == 0) {
            return 1.0;
        } else if (x == 0.0) {
            return 0.0;
        } else if (n == 1) {
            return x;
        } else if (x == 1.0) {
            return 1.0;
        } else {
            long m = n;
            boolean isNeg = m < 0;
            if (m < 0) {
                m = -m;
            }
            double half = myPow(x, (int)(m/2));
            double result = 0.0;
            if (m % 2 == 0) {
                result = half * half;
            } else {
                result = half * half * x;
            }
            if (isNeg) {
                result = 1.0/result;
            }
            return result;
        }
    }
